frontend/src/components/widgets/EventsWidget.tsx

```diff
@@ -0,0 +1,154 @@
+import { api, useBetterQuery } from "../../services/apiService";
+import { Widget } from "./Widget";
+import { TablerColor } from "../../utils/colors";
+import { useWidgetConfig } from "./widgetConfig";
+import { Loading } from "../utils/Loading";
+import { ErrorMessage } from "../utils/ErrorPage";
+import React, { useContext, useState } from "react";
+import "./widget_balance.css";
+import { Event } from "../../models/associations/event";
+import { formatDate, formatTime } from "../../utils/format";
+import dayjs  from "dayjs";
+import { Modal } from "react-bootstrap";
+import { AvatarList } from "../utils/avatar/AvatarList";
+import { EventDate } from "../associations/events/list/EventDate";
+import { UserAvatar } from "../utils/avatar/UserAvatar";
+import { Size } from "../../utils/size";
+import { UserContext } from "../../services/authService";
+
+export const EventWidget = ({ ...props }) => {
+  const { data, error, status } = useBetterQuery<Event[]>(
+    ["subscriptions.events.get"],
+    api.subscriptions.events,
+    { refetchOnWindowFocus: false }
+  );
+
+  const [selectedEvent, setSelectedEvent] = useState<Event | null>(null);
+
+  return (
+    <>
+      <Widget
+        name={"Evenements à venir"}
+        color={TablerColor.Purple}
+        bodyWrapped={false}
+        {...useWidgetConfig("events")}
+        {...props}
+      >
+        {status === "loading" ? (
+          <Loading />
+        ) : status === "error" ? (
+          <ErrorMessage>`Une erreur est apparue: ${error}`</ErrorMessage>
```

Comment:
Not 100% sure about it, but I think the backquotes and the `$` should be removed